modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustacheScriptEngineService.java

```diff
@@ -89,21 +89,19 @@ public MustacheScriptEngineService(Settings settings) {
      * */
     @Override
     public Object compile(String templateName, String templateSource, Map<String, String> params) {
-        String contentType = params.getOrDefault(CONTENT_TYPE_PARAM, JSON_CONTENT_TYPE);
-        final DefaultMustacheFactory mustacheFactory;
-        switch (contentType){
+        final MustacheFactory factory = new CustomMustacheFactory(isJsonEscapingEnabled(params));
+        Reader reader = new FastStringReader(templateSource);
+        return factory.compile(reader, "query-template");
+    }
+
+    private boolean isJsonEscapingEnabled(Map<String, String> params) {
+        switch (params.getOrDefault(CONTENT_TYPE_PARAM, JSON_CONTENT_TYPE)) {
```

Comment:
Sure, this switch statement is really ugly. I suppose my brain was in auto-pilot mode when writing this.